receiver/awsfirehosereceiver/internal/unmarshaler/cwlog/unmarshaler.go

```diff
@@ -67,95 +67,59 @@ func (u *Unmarshaler) UnmarshalLogs(compressedRecord []byte) (plog.Logs, error)
 	}
 	defer u.gzipPool.Put(r)
 
-	type resourceKey struct {
-		owner     string
-		logGroup  string
-		logStream string
+	data, err := io.ReadAll(r)
```

Comment:
Ideally we would decode directly from the reader, but I suppose this won't be worse than the existing code.

FYI we're working on extracting the unmarshaler to a separate component, and we'll optimise it there: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38821

Reply:
I suppose you are referring to this part 
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38821/files#diff-c98f13982ab112641b8e7b47af57922b415065212d6f5939a2998aee80fba34bR88-R92

Yeah that's definitely better. I'll leave the optimization to that PR then.